test/e2e/app-dir/io-is-dynamic/dynamic-io.test.ts

```diff
@@ -0,0 +1,143 @@
+import { nextTestSetup } from 'e2e-utils'
+
+process.env.__TEST_SENTINEL = 'build'
+
+describe('dynamic-io', () => {
+  const { next, isNextDev, skipped } = nextTestSetup({
+    files: __dirname,
+    skipStart: true,
+    skipDeployment: true,
+  })
+
+  if (skipped) {
+    return
+  }
+
+  beforeAll(async () => {
+    await next.start()
+    // This will update the __TEST_SENTINEL value to "run"
+    await next.render('/setenv?value=run')
```

Comment:
you should be able to provide a custom env to `nextTestSetup` rather than doing it in a route handler